docs/src/_guide/configuration.md

```diff
@@ -68,14 +68,72 @@ export <PLUGIN_NAME>_<SETTING_NAME>=<value>
 
+#### aliases
+
+Aliases allow for configuration values to be set via one of multiple keys.
+Environment variable aliases are listed next to the canonical names for the variable in the output of the [`meltano config <plugin> list`](/reference/command-line-interface#config) command.
+They can be defined via the `aliases` key in a custom plugin's `settings` configuration.
+For example, the following defines a `my_custom_username` setting with aliases `custom_tap_username` and `username`:
+
+```yaml
+# meltano.yml
+...
+plugins:
+  extractors:
+  - name: my-custom-tap
+    namespace: my_custom_tap
+    pip_url: git+https://github.com/my-organization/my-custom-tap.git
+    executable: my-custom-tap
+    capabilities:
+    - discover
+    - catalog
+    settings:
+    - name: password
+      kind: password
+    - name: my_custom_tap_username
+      aliases: [custom_tap_username, username]
+```
+
+Within a given configuration layer, a setting can be set via only a single name, whether that name is its canonical name or one of its aliases.
+So given the custom extractor defined above, the `my_custom_tap_username` setting could be set via the `MY_CUSTOM_TAP_MY_CUSTOM_TAP_USERNAME` environment variable or either the `MY_CUSTOM_TAP_CUSTOM_TAP_USERNAME` or `MY_CUSTOM_TAP_USERNAME` variables.
+But if more than one of these variables is set in the terminal environment, then an exception will be raised--even if all the relevant environment variables have the same value.
+
+To see what name or alias a setting's value is being derived from, you can run `meltano config <plugin-name> list`:
+
+```shell
+$ export MY_CUSTOM_TAP_USERNAME=some_username
+$ meltano config my-custom-tap list
+2022-06-22T10:00:00Z [info     ] Environment 'dev' is active
+password [env: MY_CUSTOM_TAP_PASSWORD] current value: 'some_very_secure_password' (from the MY_CUSTOM_TAP_PASSWORD variable in `.env`)
+my_custom_tap_username [env: MY_CUSTOM_TAP_MY_CUSTOM_TAP_USERNAME, MY_CUSTOM_TAP_CUSTOM_TAP_USERNAME, MY_CUSTOM_TAP_USERNAME] current value: 'some_username' (from the MY_CUSTOM_TAP_USERNAME variable in the environment)
+```
+
+If a setting's value is being set via multiple environment variables, the resulting error message will list the environment variables where it is being set:
+
+```shell
+$ export MY_CUSTOM_TAP_USERNAME=some_username
+$ export MY_CUSTOM_TAP_CUSTOM_TAP_USERNAME=some_username
+$ meltano config my-custom-tap
+Setting value set via multiple environment variables: ['MY_CUSTOM_TAP_CUSTOM_TAP_USERNAME', 'MY_CUSTOM_TAP_USERNAME']
+```
+
+If the values for the multiple environment variables differ, the error message will also list what the values are:
+
+```shell
+$ export MY_CUSTOM_TAP_USERNAME=some_username
+$ export MY_CUSTOM_TAP_CUSTOM_TAP_USERNAME=some_other_username
+$ meltano config my-custom-tap
+Conflicting values for setting found in: {'MY_CUSTOM_TAP_CUSTOM_TAP_USERNAME': 'some_other_username', 'MY_CUSTOM_TAP_USERNAME': 'some_username'}
```

Comment:
Do we need to print the actual values to the command line? What if this is for a password and we've just printed it out into logs potentially? It would probably suffice that we just say they are in conflict, right?